aws/aws-toolkit-jetbrains
# Fix exceptions while generating projects and on run config validation

## Types of changes

## Description
- Fix exception on startup while gathering SAM CLI information; replace synchronous execution on EDT with modifying project model after project is initialized
- Fix validation for template.yaml file to be able to call SAM deploy action
- Fix sync calls on run configuration validation to use async calls
- Add tests to verify SamVersion evaluation and getFileInfo logic

## Motivation and Context
Async calls on startup and run configuration could freeze UI thread. Move all heavy logic to async tasks to make UI responsible.

## Related Issue(s)
None

## Testing
Environment:
```
Mac OS Mojave 10.14.5 (18F132)
```
- Check exceptions on creating new project from SAM Template
- Check on open existing templates
- Check on creating sub-modules with SAM project
- Check validation for creating run configuration and check for existing run configurations.

## Screenshots (if appropriate)
None.

## Checklist

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.
